experimental/packages/opentelemetry-instrumentation-http/src/types.ts

```diff
@@ -80,6 +80,10 @@ export interface StartOutgoingSpanCustomAttributeFunction {
   (request: RequestOptions): SpanAttributes;
 }
 
+export interface CustomMetricAttributeFunction {
+  (): MetricAttributes;
```

Comment:
I believe it can be useful to have parameters like request and response in `CustomMetricAttributeFunction`, like what `HttpCustomAttributeFunction` does.

Reply:
Yea, hmm... I tried and it was too hard. Not sure how to do it within `_outgoingRequestFunction` since there is no "request" or "response"; those come much later down in a callback (within `return context.with(requestContext, () => {`). Doing this would require quite a large refactor.

So my take is that since it is (API wise) easy to add these parameters later (it would be backwards compatible with not taking aarguments). It would make this PR significantly smaller and not introduce new complexity we would rather avoid at this point.